app/routes/dog_routes.py

```diff
@@ -8,12 +8,12 @@ def __init__(self, id, name, breed, tricks=None):
         self.name = name
         self.breed = breed
         if not tricks:
-            tricks = []
+            self.tricks = []
```

Comment:
If `tricks` is `None`, we'll assign `self.tricks` to an empty list here, but then immediately overwrite it with `None` when the line below `self.tricks = tricks` runs.  Maybe we replace the self.tricks assignment with null-coalescing?
```
self.tricks = tricks or []
```